src/CompletionProvider.ts

```diff
@@ -11,20 +11,136 @@ import {
 } from './utils';
 import type {CamelCaseValues} from './utils';
 
-// check if current character or last character is .
-function isTrigger(line: string, position: Position): boolean {
+export const COMPLETION_TRIGGERS = ['.', '[', '"', "'"] as const;
+
+type FieldOptions = {
+    wrappingBracket: boolean;
+    startsWithQuote: boolean;
+    endsWithQuote: boolean;
+};
+
+/**
+ * check if current character or last character is any of the completion triggers (i.e. `.`, `[`) and return it
+ *
+ * @see COMPLETION_TRIGGERS
+ */
+function findTrigger(line: string, position: Position): string | undefined {
     const i = position.character - 1;
-    return line[i] === '.' || (i > 1 && line[i - 1] === '.');
+
+    for (const trigger of COMPLETION_TRIGGERS) {
+        if (line[i] === trigger) {
+            return trigger;
+        }
+        if (i > 1 && line[i - 1] === trigger) {
+            return trigger;
+        }
+    }
+
+    return undefined;
 }
 
-function getWords(line: string, position: Position): string {
+/**
+ * Given the line, position and trigger, returns the identifier referencing the styles spreadsheet and the (partial) field selected with options to help construct the completion item later.
+ *
+ * @returns
+ */
+function getWords(
+    line: string,
+    position: Position,
+    trigger: string,
+): [string, string, FieldOptions?] | undefined {
```

Comment:
nit: the same about return type
```suggestion
): [string, string, FieldOptions?] | void {
```